azext_edge/edge/providers/orchestration/resources/connector/opcua/certs.py

```diff
@@ -266,14 +318,137 @@ def client_add(
+        target_secretsync = target_secretsync[0]
+        secret_mapping = target_secretsync.get("properties", {}).get("objectSecretMapping", [])
+        secret_to_remove = []
+        for name in certificate_names:
+            if name not in [mapping["targetKey"] for mapping in secret_mapping]:
+                logger.warning(
+                    f"Certificate {name} not found in secretsync resource {secretsync_name}. "
+                    "Skipping removal..."
+                )
+            else:
+                # append corresponding "sourcePath" of matching "targetKey"
+                secret_to_remove.append(
+                    [mapping["sourcePath"] for mapping in secret_mapping if mapping["targetKey"] == name][0]
+                )
+
+        if not secret_to_remove:
+            raise InvalidArgumentValueError("Please provide valid certificate name(s) to remove.")
+
+        # check if OPCUA_SPC_NAME spc exists
+        target_spc = self.instances.find_existing_resources(
+            cl_resources=cl_resources,
+            resource_type=SPC_RESOURCE_TYPE,
+            resource_name=OPCUA_SPC_NAME,
+        )
+
+        if not target_spc:
+            raise ResourceNotFoundError(f"Secret Provider Class resource {OPCUA_SPC_NAME} not found.")
+
+        # get properties from default spc
+        target_spc = target_spc[0]
+        spc_properties = target_spc.get("properties", {})
+        spc_keyvault_name = spc_properties.get("keyvaultName", "")
+
+        # remove secret reference from secret sync
+        modified_secret_sync = self._remove_secrets_from_secret_sync(
+            name=secretsync_name,
+            secrets=secret_to_remove,
+            secret_sync=target_secretsync,
+            resource_group=resource_group
+        )
+
+        self._remove_secrets_from_spc(
+            secrets=secret_to_remove,
+            spc=target_spc,
+            resource_group=resource_group
+        )
+
+        if include_secrets:
+            # get keyvault client
+            self.keyvault_client = get_keyvault_client(
+                subscription_id=self.default_subscription_id,
+                keyvault_name=spc_keyvault_name,
+            )
+            # verify the behaviour of non existed secret
+            secrets: PageIterator = self.keyvault_client.list_properties_of_secrets()
+            secret_names = [secret.id for secret in secrets]
+
+            # remove secret from keyvault
+            for name in secret_to_remove:
+                # perform delete operation if name exists in secret_names(endwith)
+                if any(secret_name.endswith(name) for secret_name in secret_names):
+                    with console.status(f"Deleting and purging secret {name} from keyvault {spc_keyvault_name}..."):
+                        poller = self.keyvault_client.begin_delete_secret(name)
+                        wait_for_terminal_state(poller)
+                        self.keyvault_client.purge_deleted_secret(name)
```

Comment:
Is purge_deleted_secret an LRO ?

Reply:
purge is not LRO

Reply:
FYI https://learn.microsoft.com/en-us/python/api/azure-keyvault-secrets/azure.keyvault.secrets.secretclient?view=azure-python#azure-keyvault-secrets-secretclient-purge-deleted-secret